File: src/LeetCode/Single Number.py
"""
https://leetcode.com/explore/challenge/card/30-day-leetcoding-challenge/528/week-1/3283/
A ^ A = 0, A ^ B = B ^ A
"""
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        answer = nums[0]
        for i in range(1, len(nums)):
            answer ^= nums[i]
        return answer

File: src/LeetCode/Single Number.java
/**
 * https://leetcode.com/problems/single-number/
 * Using bit mask
 */
//1. Simple Solution (1ms)
class Solution {
    public int singleNumber(int[] nums) {
        int ans = 0;
        for (int n : nums) {
            ans ^= n;
        }
        return ans;
    }
}

//2. Solution using stream (5ms)
class Solution {
    public int singleNumber(int[] nums) {
        return Arrays.stream(nums).reduce(0, (a, b) -> a ^ b);
    }
}
